src/server.js

```diff
@@ -35,10 +37,22 @@ export default function start(
 
 	const baseRoutes = getRoutes(renderRequestMap);
 	const finalRoutes = [...routes, ...baseRoutes];
+	let tls = null;
+	if (
+		appConfig.app_server.protocol === 'https' &&
+		appConfig.app_server.key_file &&
+		appConfig.app_server.crt_file
```

Comment:
sounds good. Unfortunately it looks like the custom validation callbacks defined by node-convict only take the current `value` as an argument, so I don't see a built-in way of validating interdependent values, but it's easy enough to do it outside of `config.validate()`. I'm considering whether to submit a PR to node-convict to add that functionality, though - seems like it would be pretty useful.